### ISSUE TITLE
[Bug]  After enabling replication, the number of topic partitions cannot be modified

### ISSUE THREAD

**Issue by tianshimoyi** [apache/pulsar#21049]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Version

v2.9.4

### Minimal reproduce step

1. Turn on geo-replication according to the official document steps [https://pulsar.apache.org/docs/2.11.x/administration-geo/]
2. Master and slave clusters use different zookeepers
3. Modify the number of partitions of topic multiple times

### What did you expect to see?

The number of partitions is modified correctly and synchronized to another cluster

### What did you see instead?

1. Failed to modify the number of partitions
![截屏2023-08-22 下午5 56 11](https://github.com/apache/pulsar/assets/68532999/7a322e4b-a3d3-432a-a197-654b67dc307e)

2. Another cluster has an incorrect number of partitions
![截屏2023-08-22 下午5 57 14](https://github.com/apache/pulsar/assets/68532999/59b75560-7df6-40f4-8c9d-a1ac21daa2dc)

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by mattisonchao** [comment]
Hi @tianshimoyi 

You should check why `t1-partition-5` and `t1-partition-4` has been created and if they can be discarded.

If you can discard `t1-partition-5` and `t1-partition-4` you can try using `delete topic persistnet://tcloud/backup-ns2/t1-partition-5` and `delete topic persistnet://tcloud/backup-ns2/t1-partition-4` to delete these partitions.
Afterwards, you can expand the partitions by admin.

_**If `partition 4/5` is useful and some data exist there. you can try updating the partition by `force` flag.**_

**Comment by tianshimoyi** [comment]
@mattisonchao The reason why t1-partition-5 and t1-partition-4 were created is that I expanded the partition in a cluster, and the synchronized cluster also expanded automatically, but the topic meta information did not change. After several operations to expand the partition, there is no way Expanded, and reported the above error.

**Comment by mattisonchao** [comment]
Hi, @tianshimoyi 

Because the topic partition meta didn't update, I think the partition-4 and partition-5 also don't have data. Because the client needs to fetch metadata before publishing, maybe you can choose to delete them and update the partition number again.

**Comment by mattisonchao** [comment]
If you don't wanna force update the partition number to bypass the check. you can even modify the zookeeper node data directly by zk shell. that is a json data which maybe not be hard. :)

**Comment by tianshimoyi** [comment]
@mattisonchao Yes, I know how to modify. But why is this the case? Is this a bugger?

**Comment by mattisonchao** [comment]
It seems like a bug before this [PR](https://github.com/apache/pulsar/pull/19166/files). The broker will try to create the partition first and then update the metadata.

**Comment by mattisonchao** [comment]
Would you mind to upgrade your cluster to TLS 3.0?
